File: feature/keyboard/contract/src/main/java/net/twisterrob/colorfilters/android/keyboard/KeyboardHandler.java
package net.twisterrob.colorfilters.android.keyboard;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.EditText;

public interface KeyboardHandler {

	interface CustomKeyboardListener {
		void customKeyboardShown();

		void customKeyboardHidden();
	}

	void hideCustomKeyboard();

	void showCustomKeyboard(@NonNull View v);

	void registerEditText(@NonNull EditText editText);

	void unregisterEditText(@NonNull EditText editText);

	void setHapticFeedback(boolean haptic);

	boolean handleBack();

	void setCustomKeyboardListener(@NonNull CustomKeyboardListener listener);
}

File: feature/keyboard/contract/src/main/kotlin/net/twisterrob/colorfilters/android/keyboard/KeyboardHandler.kt
package net.twisterrob.colorfilters.android.keyboard

import android.view.View
import android.widget.EditText

interface KeyboardHandler {

	interface CustomKeyboardListener {

		fun customKeyboardShown()

		fun customKeyboardHidden()
	}

	fun hideCustomKeyboard()

	fun showCustomKeyboard(v: View)

	fun registerEditText(editText: EditText)

	fun unregisterEditText(editText: EditText)

	var hapticFeedback: Boolean

	fun handleBack(): Boolean

	var customKeyboardListener: CustomKeyboardListener?
}
